src/lib/cxplat.user.vcxproj

```diff
@@ -1,71 +1,32 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project DefaultTargets="Build" ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="..\cxplat.props" />
-  <ItemGroup>
-    <ClCompile Include="cxplat_winuser.c" />
-  </ItemGroup>
+  <PropertyGroup>
+    <ProjectGuid>{702A7493-A6A4-4209-AB69-E1E4D1EB0FB7}</ProjectGuid>
+    <TargetName>platformuser</TargetName>
+    <UndockedType>lib</UndockedType>
+    <UndockedDir>$(SolutionDir)submodules\undocked\</UndockedDir>
+    <UndockedOut>$(SolutionDir)artifacts\</UndockedOut>
```

Comment:
How does the output folder structure look like? When it's built by XDP, it has to match the XDP artifact folder structure so that XDP project can find the lib.

Reply:
I just tried XDP out locally and it seems to work. But I will send a draft XDP PR first, pointing to this branch to see if its CI passes.

Reply:
https://github.com/microsoft/xdp-for-windows/pull/603

Reply:
I downloaded an artifact archive from the workflow. It doesn't look like the cxplat.lib is placed in the bin folder.

```
PS C:\Users\huanyi\Downloads\Release-windows-windows-2022-x64-vs> tree /F
Folder PATH listing for volume Windows
Volume serial number is 169F-C81F
C:.
├───bin
│   └───amd64fre
│           platformuser.lib
│
├───corenet-ci-main
│   │   .gitignore
│   │   CODE_OF_CONDUCT.md
│   │   LICENSE
│   │   README.md
│   │   SECURITY.md
│   │   SUPPORT.md
│   │
│   ├───query
│   │       dpt-summary.ps1
│   │       dpt.ps1
│   │       readme.md
│   │
│   └───vm-setup
│       │   azp_vm_setup.ps1
│       │   copysymbols.cmd
│       │   CoreNet-CI-Startup.ps1
│       │   CoreNetSign.pfx
│       │   CoreNetSignRoot.cer
│       │   devcon.exe
│       │   dswdevice.exe
│       │   Generate-Certificates.ps1
│       │   kd.exe
│       │   livekd64.exe
│       │   notmyfault64.exe
│       │   procdump64.exe
│       │   refreshazpvms.ps1
│       │   sfpcopy.exe
│       │   testroot-sha2.cer
│       │   tracepdb.exe
│       │   unattend.xml
│       │   vs_BuildTools.exe
│       │   wsario.exe
│       │
│       ├───duonic
│       │       dswdevice.exe
│       │       duonic.cat
│       │       duonic.inf
│       │       duonic.ps1
│       │       duonic.sys
│       │
│       └───sdk
│               Inf2Cat.exe
│               inf2cat.exe.manifest
│               Microsoft.UniversalStore.HardwareWorkflow.Cabinets.dll
│               Microsoft.UniversalStore.HardwareWorkflow.Catalogs.dll
│               Microsoft.UniversalStore.HardwareWorkflow.InfReader.dll
│               Microsoft.UniversalStore.HardwareWorkflow.SubmissionBuilder.dll
│               WindowsProtectedFiles.xml
│
└───obj
    └───amd64fre
        └───cxplat.user
            │   build.lastcodeanalysissucceeded
            │   cxplat.user.vcxproj.FileListAbsolute.txt
            │   cxplat_winuser.nativecodeanalysis.xml
            │   cxplat_winuser.obj
            │   NativeCodeAnalysis.read.1.tlog
            │   platformuser.lib.recipe
            │   vc.nativecodeanalysis.all.xml
            │
            └───cxplat.user.tlog
                    CL.command.1.tlog
                    Cl.items.tlog
                    CL.read.1.tlog
                    CL.write.1.tlog
                    cxplat.user.lastbuildstate
                    Lib-link.read.1.tlog
                    Lib-link.write.1.tlog
                    Lib.command.1.tlog
```

Reply:
oh, it's platformuser.lib now.

Reply:
Why change the name to platformuser though? cxplat.user.lib or winuser\cxplat.lib would be a better name.

Reply:
I thought I copied all the previous names. I didn't mean to change anything. Are you sure I changed it?

Reply:
yes, target name used to be ```<TargetName>cxplat</TargetName>```

Reply:
Fixed

Reply:
You might have to name it something different or add a prefix to the out path. Otherwise, the user mode lib will have the same path with the kernel lib.

Reply:
How'd that work before? I'm not trying to change any names here.

Reply:
Well, the CI all passed, so maybe it's not needed?

Reply:
Before, the output path is different. It has one more level (winuser and winkerenl).

The CI passed because it won't build both at the same time but we might do it in xdp.

I think we just need to add that one extra level. You wouldn't even need to change xdp then.